### PR TITLE
noImplicitAny: Down approx 200 errors

### PR DESCRIPTION
**Pull request by tskarhed**
#14714

### CODE REVIEW

File: public/app/features/panel/panel_header.ts [hunk 1]
```diff
@@ -12,7 +12,10 @@ const template = `
   <span class="panel-time-info" ng-if="ctrl.timeInfo"><i class="fa fa-clock-o"></i> {{ctrl.timeInfo}}</span>
 </span>`;
 
-function renderMenuItem(item, ctrl) {
+function renderMenuItem(
+  item: { divider: any; submenu: any; click: any; href: any; icon: any; text: any; shortcut: any },
```
Comment (torkelo) [inline public/app/features/panel/panel_header.ts]: there is a PanelMenuItem type from @grafana/ui you can use

File: public/app/features/plugins/built_in_plugins.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ import * as barGaugePanel from 'app/plugins/panel/bargauge/module';
 
 import * as exampleApp from 'app/plugins/app/example-app/module';
 
-const builtInPlugins = {
+const builtInPlugins: any = {
```
Comment (torkelo) [inline public/app/features/plugins/built_in_plugins.ts]: not sure you need this, typescript should be able to figure out type from declaration?

File: public/app/features/plugins/datasource_srv.ts [hunk 3]
```diff
@@ -108,7 +115,7 @@ export class DatasourceSrv implements DataSourceService {
     return sources;
   }
 
-  getMetricSources(options?) {
+  getMetricSources(options?: { skipVariables?: any }) {
```
Comment (torkelo) [inline public/app/features/plugins/datasource_srv.ts]: skipVariables: boolean  I think

File: public/app/features/plugins/datasource_srv.ts [hunk 4]
```diff
@@ -148,7 +155,7 @@ export class DatasourceSrv implements DataSourceService {
     return metricSources;
   }
 
-  addDataSourceVariables(list) {
+  addDataSourceVariables(list: any) {
```
Comment (torkelo) [inline public/app/features/plugins/datasource_srv.ts]: we have a Variable interface defined in app/types/templates.ts

**Review by torkelo — APPROVED** [review]
Good, soon under 1000? :) Found some minor things

File: public/app/features/panel/panel_header.ts (hunk 1 above)
Comment (dehrax) [reply public/app/features/panel/panel_header.ts]: Not sure how to handle this. The properties being used are `divider, submenu, click, href, icon` while  `PanelMenuItem` look like
```typescript

export interface PanelMenuItem {
  type?: 'submenu' | 'divider';
  text?: string;
  iconClassName?: string;
  onClick?: () => void;
  shortcut?: string;
  subMenu?: PanelMenuItem[];
}
```
Is the right way to go to add the fields being used to the type, or should the code be rewritten to match the type (which will probably break things)?

File: public/app/features/plugins/built_in_plugins.ts (hunk 2 above)
Comment (dehrax) [reply public/app/features/plugins/built_in_plugins.ts]: That's what I thought, but as it turns out noImplicitAny complains when trying to read an object property by passing a string as index `builtInPlugins[propertyString]`. The other option would be to do ts-ignore, but it feels dirty.

File: public/app/features/panel/panel_header.ts (hunk 1 above)
Comment (torkelo) [reply public/app/features/panel/panel_header.ts]: Ok, can you create a new type named AngularPanelMenuItem for this then?

File: public/app/features/plugins/built_in_plugins.ts (hunk 2 above)
Comment (torkelo) [reply public/app/features/plugins/built_in_plugins.ts]: Ok, let’s use any then

File: public/app/features/plugins/datasource_srv.ts (hunk 4 above)
Comment (dehrax) [reply public/app/features/plugins/datasource_srv.ts]: This also uses the properties `value, meta, sort` that are not on the Variable interface. Should I add those to the interface?

**Pull request merged**